pkg/macvlan/macvlan_test.go

```diff
@@ -0,0 +1,134 @@
+package macvlan
+
+import (
+	"fmt"
+	"testing"
+
+	egresstest "github.com/openshift/egress-router-cni/pkg/testing"
+	netlink_mocks "github.com/openshift/egress-router-cni/pkg/testing/mocks/github.com/vishvananda/netlink"
+	util "github.com/openshift/egress-router-cni/pkg/util"
+	util_mocks "github.com/openshift/egress-router-cni/pkg/util/mocks"
+	"github.com/vishvananda/netlink"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFillNetConfDefaults(t *testing.T) {
+	mockNetLinkOps := new(util_mocks.NetLinkOps)
+	mockLink := new(netlink_mocks.Link)
+	// below sets the `netLinkOps` in util/net_linux.go to a mock instance for purpose of unit tests execution
+	util.SetNetLinkOpMockInst(mockNetLinkOps)
+
+	tests := []struct {
+		desc             string
+		inpNetConf       *NetConf
+		inpClusterConf   *ClusterConf
```

Comment:
Just `clusterConf`

Reply:
will change it

Reply:
the inp and out are used as prefixes to differentiate between fields that is either input to function of ouput from the function.